nodeadm/internal/system/networking.go

```diff
@@ -89,11 +96,79 @@ func (a *networkingAspect) ensureEKSNetworkConfiguration(cfg *api.NodeConfig) er
 	return nil
 }
 
+func (a *networkingAspect) ensureMulticardNetworkConfiguration(cfg *api.NodeConfig) error {
+	var networkRestartRequired bool
+	routeTableId := 1001
+	routePriority := 32765
+
+	for _, card := range cfg.Status.Instance.NetworkCards {
+		if card.CardIndex == 0 {
+			continue
+		}
+
+		networkInterfaceConfName := fmt.Sprintf("80-card%d.network", card.CardIndex)
+		networkInterfaceConfPathName := fmt.Sprintf("%s/%s", administrationNetworkDir, networkInterfaceConfName)
+
+		if exists, err := util.IsFilePathExists(networkInterfaceConfPathName); err != nil {
+			return fmt.Errorf("failed to check configuration existance for %s: %w", networkInterfaceConfName, err)
+		} else if exists {
+			zap.L().Sugar().Infof("%s already exists, skipping configuration", networkInterfaceConfName)
+			continue
+		}
+
+		templateVars := networkInterfaceTemplateVars{
+			PermanentMACAddress: card.MAC,
+			IPAddress:           card.IpAddress,
+			RouteTableId:        int16(routeTableId),
+			RoutePriority:       int16(routePriority),
+		}
+		routeTableId++
+		routePriority--
+
+		interfaceConfigContent, err := a.generateNetworkConfigFile(networkInterfaceConfName, templateVars)
+		if err != nil {
+			return fmt.Errorf("failed to generate %s configuration: %w", networkInterfaceConfName, err)
+		}
+
+		if err := os.WriteFile(networkInterfaceConfPathName, interfaceConfigContent, networkConfFilePerms); err != nil {
+			return fmt.Errorf("failed to write %s configuration: %w", networkInterfaceConfName, err)
+		}
+		zap.L().Sugar().Infof("Multicard instance found, configuring card with index: %d, network file: %s", card.CardIndex, networkInterfaceConfPathName)
+		networkRestartRequired = true
+	}
+
+	if networkRestartRequired {
```

Comment:
Nit: maybe we should combine the "reloadNetworkConfigurations" with `ensureEKSNetworkConfiguration` to do reload once.

Reply:
I can restart the network in the Setup function, after we configure both primary and multicard interfaces.